### PR TITLE
Adding Future Extension to track completion of a Future array

### PR DESCRIPTION
**Pull request by kneekey23**
*Description of changes:* Added single extension function on a group of futures

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Source/AwsCommonRuntimeKit/crt/Future.swift [hunk 1]
```diff
@@ -91,4 +91,36 @@ extension Future {
...
+                }
+            case .failure(let error):
+                future.fail(error)
+            }
+        }
+
+        for(index, future) in futures.enumerated() {
+            if let result = future._value {
+                processResult(index, result)
+
```
Comment (justinboswell) [inline Source/AwsCommonRuntimeKit/crt/Future.swift:121]: weird newline

File: Source/AwsCommonRuntimeKit/crt/Future.swift [hunk 2]
```diff
@@ -91,4 +91,36 @@ extension Future {
 
         return promise
     }
+
+    static func whenAllComplete(_ futures: [Future<Value>]) -> Future<Void> {
```
Comment (justinboswell) [inline Source/AwsCommonRuntimeKit/crt/Future.swift:95]: Should this be `Future<Value>` instead of `Future<Void>`?

Comment (justinboswell) [reply Source/AwsCommonRuntimeKit/crt/Future.swift:95]: Seems like you'd want the value if the bundled futures produce one.

Comment (kneekey23) [reply Source/AwsCommonRuntimeKit/crt/Future.swift:95]: ya I thought about returning each value in a completion as they come in, but I wrote this for a specific use case. I'll make it more generic :) good point

**Review by justinboswell — APPROVED** [review]
Based on discussion in slack, going to ignore the value return for now.

File: Source/AwsCommonRuntimeKit/crt/Future.swift (hunk 2 above)
Comment (kneekey23) [reply Source/AwsCommonRuntimeKit/crt/Future.swift:104]: leaving off for now and add documentation. will return to this if required

**Pull request merged**